# Chapter 10 - Gravitational-wave Bursts detection: Methods

**Note**: In this set of challenges you will have to download a file. You can do it manually by saving it on your computer and uploading to Colab, but you have to re-do this everytime you relaunch your Colab session (unless you link it with your Google Drive space and save the files there). An useful alternative is to download the file directly in the notebook, as we do for gravitational-wave data from the Gravitational Wave Open Science Center.

For instance, you can do:

In [1]:
import numpy as np

In [2]:
# numpy's genfromtxt can download directly, you just need to get the link:
data=np.genfromtxt('https://raw.githubusercontent.com/astrophysicsvivien/GWA2021/main/chapter10/correlation_test.txt.gz')
data

array([[ 5.62178e-21,  1.83831e-19],
       [-1.45019e-19, -5.19038e-21],
       [ 5.58187e-20,  5.68165e-20],
       ...,
       [-1.00425e-19, -7.41446e-20],
       [ 3.96440e-19,  1.37329e-19],
       [ 1.43393e-19,  8.58192e-20]])

In [3]:
# or you can download the file in the notebook, and read it in in the standard way:
! wget -q https://raw.githubusercontent.com/astrophysicsvivien/GWA2021/main/chapter10/correlation_test.txt.gz
data2=np.genfromtxt('correlation_test.txt.gz')
data2

array([[ 5.62178e-21,  1.83831e-19],
       [-1.45019e-19, -5.19038e-21],
       [ 5.58187e-20,  5.68165e-20],
       ...,
       [-1.00425e-19, -7.41446e-20],
       [ 3.96440e-19,  1.37329e-19],
       [ 1.43393e-19,  8.58192e-20]])

## Detecting a Supernova signal using cross-correlation

The file [correlation_test.txt.gz](https://raw.githubusercontent.com/astrophysicsvivien/GWA2021/main/chapter10/correlation_test.txt.gz) contains two columns of data:

1. data from detector 1 “H”
2. data from detector 2 “L”

The sampling rate is 16384 sec^-1, and the total data duration is 16 sec. The noise is white 
Gaussian with standard deviation 1e-19. Somewhere in the data is a GW signal from a 
core-collapse supernova, similar to that in the lecture notes. By cross-correlating data from 
H and L in 1/8 sec chunks, find the time of arrival of the supernova signal to the nearest +/-1/16 sec.
Determine the difference of arrival times of the signal in H and L to the nearest millisecond.

## Using the Null Energy to reject glitches

The file [null_test.txt.gz](https://raw.githubusercontent.com/astrophysicsvivien/GWA2021/main/chapter10/null_test.txt.gz) contains three columns of data:
1. data from detector 1 “H”
2. data from detector 2 “L”
3. data from detector 3 “V”

The sampling rate is 16384 sec^-1, and the total data duration is 2 sec. The noise is white 
Gaussian with standard deviation 1e-19. 

This data was taken around the time of an observed supernova, and the data streams have 
been aligned in time so that any GW from the supernova should be simultaneous in the three
data streams. Unfortunately, the detectors were glitching at this time and there are two events
visible in the data, at t = 0.5 sec after the data start and at t = 1.5 sec. 

By computing the null energy, show that the first transient is a noise glitch and that the second 
transient is consistent with a GW. You may use the following antenna response values for the 
sky position of the supernova:
```
     H1        L1      V1 
F_+  -0.3609   0.4353   0.5536
F_x  -0.2401   0.1514  -0.1888
```

## Reconstructing the GW signal

The file [waveform_reconstruction_test.txt.gz](https://raw.githubusercontent.com/astrophysicsvivien/GWA2021/main/chapter10/waveform_reconstruction_test.txt.gz) contains one column of data:
1. h_+ polarisation of the simulated GW signal from the null energy example above

In this simulation the GW signal had h_x = 0.

By applying the Moore-Penrose inverse to the data from the previous example around time 
t = 1.5 sec, compute the maximum-likelihood estimate of the GW plus polarisation. You may 
wish to compare your estimate to the actual waveform used, in the attached file.